internal/subscription/core.go

```diff
@@ -497,3 +501,60 @@ func (c *Core) Migrate(ctx context.Context, names []string) error {
 	logger.Ctx(ctx).Infow("migration: request completed.", "subscriptionsUpdated", updatedSubCount)
 	return nil
 }
+
+func (c *Core) deleteSubscriptionTopics(ctx context.Context, m *Model) error {
+	if m == nil || m.GetTopic() == "" {
```

Comment:
Why do we have to write all delete logic here? Can we separate it into a different struct or helper?

Reply:
@subin this is helper function only, we cannot separate this logic anyway as we need to delete all subscription related topic here i am passing only name of different topic and logic to delete from broker is written on different method only.

Reply:
changed in the new commit